File: example-databinding/src/main/java/com/xwray/groupie/example/databinding/item/LiveDataItem.java
package com.xwray.groupie.example.databinding.item;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import com.xwray.groupie.databinding.BindableItem;
import com.xwray.groupie.example.databinding.R;
import com.xwray.groupie.example.databinding.databinding.ItemLiveDataBinding;

public class LiveDataItem extends BindableItem<ItemLiveDataBinding> {

    // Simple wrapper around liveData
    public class DataWrapper {
        private MutableLiveData<String> data = new MutableLiveData<>();

        public MutableLiveData<String> getData() {
            return data;
        }
    }

    public DataWrapper internalDataWrapper = new DataWrapper();

    public LiveDataItem() {
        super();
    }

    @Override public int getLayout() {
        return R.layout.item_live_data;
    }

    @Override public void bind(@NonNull ItemLiveDataBinding viewBinding, int position) {
        viewBinding.setWrapper(internalDataWrapper);
    }
}

File: example/src/main/java/com/xwray/groupie/example/item/LiveDataItem.kt
package com.xwray.groupie.example.item

import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer
import com.xwray.groupie.example.R
import com.xwray.groupie.kotlinandroidextensions.Item
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.item_live_data.*

class LiveDataItem(private val data: LiveData<String>) : Item() {

    override fun bind(viewHolder: ViewHolder, position: Int) {
        data.observe(viewHolder, Observer {
            viewHolder.text.text = it
        })
    }

    override fun getLayout(): Int = R.layout.item_live_data
}
